src/libostree/ostree-repo-commit.c

```diff
@@ -2017,6 +2017,207 @@ ostree_repo_write_commit_detached_metadata (OstreeRepo      *self,
+  return g_file_resolve_relative_path (self->objects_dir, buf);
+}
+
+gboolean
+_ostree_repo_read_commit_compat_signature (OstreeRepo      *self,
+                                           const char      *checksum,
+                                           GBytes         **out_signature,
+                                           GCancellable    *cancellable,
+                                           GError         **error)
+{
+  gboolean ret = FALSE;
+  gs_unref_object GFile *signature_path =
+    _ostree_repo_get_commit_compat_signature_loose_path (self, checksum);
+  GMappedFile *signature_file = NULL;
+  gs_unref_bytes GBytes *ret_signature = NULL;
+  GError *temp_error = NULL;
+
+  signature_file = gs_file_map_noatime (signature_path, cancellable,
+                                        &temp_error);
+  if (!signature_file)
+    {
+      if (g_error_matches (temp_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        {
+          g_clear_error (&temp_error);
+        }
+      else
+        {
+          g_prefix_error (error, "Unable to read existing compat signature: ");
+          g_propagate_error (error, temp_error);
+          goto out;
+        }
+    }
+  else
+    {
+      ret_signature = g_mapped_file_get_bytes (signature_file);
+    }
+
+  ret = TRUE;
+  ot_transfer_out_value (out_signature, &ret_signature);
+ out:
+  if (signature_file)
+    g_mapped_file_unref (signature_file);
+  return ret;
+}
+
+gboolean
+_ostree_repo_write_commit_compat_signature (OstreeRepo      *self,
+                                            const char      *checksum,
+                                            GBytes          *signature,
+                                            GCancellable    *cancellable,
+                                            GError         **error)
+{
+  gboolean ret = FALSE;
+  gs_unref_object GFile *signature_path =
+    _ostree_repo_get_commit_compat_signature_loose_path (self, checksum);
+  gsize signature_size;
+  gconstpointer signature_buf = NULL;
+
+  signature_buf = g_bytes_get_data(signature, &signature_size);
+  if (signature_size == 0)
+    {
+      GError *local_error = NULL;
+
+      (void) g_file_delete (signature_path, cancellable, &local_error);
+
+      if (g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        {
+          g_clear_error (&local_error);
+        }
+      else if (local_error != NULL)
+        {
+          g_propagate_error (error, local_error);
+          g_prefix_error (error, "Unable to delete compat signature: ");
+          goto out;
+        }
+    }
+  else
+    {
+      if (!g_file_replace_contents (signature_path,
+                                    signature_buf,
+                                    signature_size,
+                                    NULL, FALSE, 0, NULL,
+                                    cancellable, error))
+        {
+          g_prefix_error (error, "Failed to write compat signature: ");
```

Comment:
I don't think so. The error is prefixed with another message, but it's the same error set in the failing call. This is pretty topical in ostree code.